core/src/test/java/org/apache/brooklyn/core/mgmt/rebind/RebindWithDeserializingClassRenamesTest.java

```diff
@@ -18,76 +18,134 @@
+    }
+
+    private Map<ConfigKey<?>, Object> getEntityConfig(final String entityId) {
+        final EntityInternal entity = (EntityInternal) mgmt().getEntityManager().getEntity(entityId);
+        assertThat(entity).isNotNull();
+        return entity.config().getAllLocalRaw();
+    }
 
-        // Check the config keys are as expected 
-        EntityInternal entity = (EntityInternal) mgmt().getEntityManager().getEntity(entityId);
-        Map<ConfigKey<?>, Object> config = entity.config().getAllLocalRaw();
-        ConfigKey<?> keyWithInheritanceNone = Iterables.find(config.keySet(), ConfigPredicates.nameEqualTo("my.config.inheritanceNone"));
-        ConfigKey<?> keyWithInheritanceAlways = Iterables.find(config.keySet(), ConfigPredicates.nameEqualTo("my.config.inheritanceAlways"));
-        ConfigKey<?> keyWithInheritanceMerged = Iterables.find(config.keySet(), ConfigPredicates.nameEqualTo("my.config.inheritanceMerged"));
-        
-        assertConfigRuntimeInheritanceMode(keyWithInheritanceNone, BasicConfigInheritance.NOT_REINHERITED);
-        assertConfigRuntimeInheritanceMode(keyWithInheritanceAlways, BasicConfigInheritance.OVERWRITE);
-        assertConfigRuntimeInheritanceMode(keyWithInheritanceMerged, BasicConfigInheritance.DEEP_MERGE);
+    private ConfigInheritance inheritanceModeByKeyName(final Map<ConfigKey<?>, Object> config, final String keyName) {
+        return inheritanceMode(configKeyByName(config, keyName));
     }
 
-    private void assertConfigRuntimeInheritanceMode(ConfigKey<?> key, ConfigInheritance expected) throws Exception {
-        ConfigInheritance val = key.getInheritanceByContext().get(InheritanceContext.RUNTIME_MANAGEMENT);
-        assertEquals(val, expected);
+    private ConfigInheritance inheritanceMode(ConfigKey<?> key) {
+        return key.getInheritanceByContext().get(InheritanceContext.RUNTIME_MANAGEMENT);
+    }
+
+    private ConfigKey<?> configKeyByName(final Map<ConfigKey<?>, Object> config, final String keyName) {
+        return config.keySet().stream().filter(ConfigPredicates.nameEqualTo(keyName)::apply).findFirst().get();
+    }
+
+    private static class Transform {
+
+        private String legacyClassName;
+        private String currentClassName;
+        private ConfigInheritance inheritanceMode;
+        private String configKeyName;
+
+        private Transform(final String legacyClassName,
+                          final String currentClassName,
+                          final ConfigInheritance inheritanceMode,
+                          final String configKeyName) {
+            this.legacyClassName = legacyClassName;
+            this.currentClassName = currentClassName;
+            this.inheritanceMode = inheritanceMode;
+            this.configKeyName = configKeyName;
+        }
+
+        static Transform of(final String legacyClassName,
+                            final ConfigInheritance configInheritance,
```

Comment:
Preference for consistent naming: the constructor calls it `inheritanceMode`, as does the field - so I'd call it that in this factory method as well.